contracts/settling_game/L02_Resources.cairo

```diff
@@ -375,15 +351,56 @@ func fetch_resource_upgrade_ids{
 
     # TODO: ADD IN DYNAMIC COST ACCORDING TO RESOURCE LEVEL
     return (
-        resource_ids=ResourceUpgradeIds(
-        resource_1=resource_1,
-        resource_2=resource_2,
-        resource_3=resource_3,
-        resource_4=resource_4,
-        resource_5=resource_5,
-        resource_1_values=resource_1_values,
-        resource_2_values=resource_2_values,
-        resource_3_values=resource_3_values,
-        resource_4_values=resource_4_values,
-        resource_5_values=resource_5_values))
+        resource_values=ResourceUpgradeValues(
+        resource_1,
+        resource_2,
+        resource_3,
+        resource_4,
+        resource_5,
+        resource_1_values,
+        resource_2_values,
+        resource_3_values,
+        resource_4_values,
+        resource_5_values))
+end
+
+func calculate_resource_level_cost{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        token_id : Uint256, resource_id : felt, resource_value : felt) -> (value : felt):
+    let (controller) = MODULE_controller_address()
+
+    # STATE
+    let (resources_state_address) = IModuleController.get_module_address(
+        controller, ModuleIds.S02_Resources)
+
+    # GET RESOURCE LEVEL
+    let (level) = IS02_Resources.get_resource_level(resources_state_address, token_id, resource_id)
+
+    # CALC
+    let value = level * resource_value
+
+    return (value)
+end
+
+func calculate_resource_output{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        token_id : Uint256, resource_id : felt) -> (value : felt):
+    alloc_locals
+    let (controller) = MODULE_controller_address()
+
+    # STATE
+    let (resources_state_address) = IModuleController.get_module_address(
+        controller, ModuleIds.S02_Resources)
+
+    # GET RESOURCE LEVEL
+    let (level) = IS02_Resources.get_resource_level(resources_state_address, token_id, resource_id)
+
+    local l
+    # CALC
+    if level == 0:
+        assert l = BASE_RESOURCES_PER_DAY
```

Comment:
You can directly `return` form the if-else block and skip the `local` which will probably save like 2 computational steps 😅